src/App.tsx

```diff
@@ -15,47 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.NONE:
+        return 0;
+
+      default:
+        throw new Error('Type is not valid');
```

Comment:
```suggestion
      case SortType.NONE:
      default:
        return 0;
```
In this case, it's better not to throw an exception, but just return 0.
An exception will cause the application to break, and a return of 0 will work fine.
Giving the user the wrong sort is better than crashing the app with an error.